restli-client/src/main/java/com/linkedin/restli/client/FluentClient.java

```diff
@@ -0,0 +1,38 @@
+package com.linkedin.restli.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * A common interface for client that implements FluentAPIs
+ */
+interface FluentClient
+{
+
+  ThreadLocal<List<ExecutionGroup>> _executionGroup = new ThreadLocal<List<ExecutionGroup>>()
+  {
+    @Override
+    public List<ExecutionGroup> initialValue()
+    {
+      return new ArrayList<ExecutionGroup>();
```

Comment:
I think using a LinkedList as stack and using push/pop/peek apis would make it clear on what is happening. Also ArrayList can result in wasted memory. Default capacity is 10 and doesn't reduce the size when elements are removed.

Reply:
ok, using it as a stack was my initial thought, or we can update to LinkedList, Let's discuss

Reply:
took your advice.